src/lsp-server.ts

```diff
@@ -1123,4 +1155,71 @@ export class LspServer {
             ...workspacePreference.inlayHints ?? {}
         };
     }
+
+    async semanticTokensFull(params: lsp.SemanticTokensParams): Promise<lsp.SemanticTokens> {
+        const file = uriToPath(params.textDocument.uri);
+        this.logger.log('semanticTokensFull', params, file);
+        if (!file) {
+            return { data: [], resultId: '' };
```

Comment:
Can you remove the `resultId` from results? We don't support deltas and according to the spec `resultId` is optional.

Reply:
I've applied that minor change + a removed a couple of unneeded type annotations. The changes look good to me. Just gonna wait for you to give an OK to my changes also.